lib/spack/spack/solver/asp.py

```diff
@@ -2463,6 +2463,37 @@ def _develop_specs_from_env(spec, env):
     spec.constrain(dev_info["spec"])
 
 
+def _is_reusable_external(packages, spec):
+    pkg_cfg = packages.get(spec.name)
+    if not pkg_cfg:
+        return False
```

Comment:
This won't catch packages that have externals defined under a virtual they provide, e.g.

```
packages:
  mpi:
    externals:
    - spec: mpich@4.1
      prefix: /path/to/prefix
```

Reply:
Didn't realize that was possible